core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -642,14 +657,7 @@ abstract class AbstractFetcherThread(name: String,
     } else if (leader.isTruncationOnFetchSupported) {
       // With old message format, `latestEpoch` will be empty and we use Truncating state
       // to truncate to high watermark.
-      val latestEpochInLog = latestEpochFromLog(tp)
-      // For mirrored partitions, use epochs from the local log (which contain source cluster epochs).
-      // For regular followers, use epochs from partition metadata (destination cluster epochs).
-      val lastFetchedEpoch: Optional[Integer] = if (initialFetchState.mirrorName.nonEmpty) {
-        if (latestEpochInLog.isPresent)
-          latestEpochInLog
-        else Optional.of(0) // this should only happen when no data in local log
-      } else latestEpoch(tp)
+      val lastFetchedEpoch = latestEpoch(tp)
```

Comment:
Because right now, the leader epoch cache will align with the log batches, we can directly get from the cache without reading the log segment.